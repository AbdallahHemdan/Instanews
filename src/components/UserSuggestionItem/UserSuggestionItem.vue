<template>
  <div class="suggestions-item">
    <div class="suggestions-item__header">
      <div class="suggestions-item__remove">&times;</div>
    </div>

    <div class="suggestions-item__body">
      <div class="suggestions-item__img-container">
        <img
          :src="suggestion.img"
          :alt="suggestion.name"
          draggable="false"
          class="suggestions-item__img"
        />
      </div>

      <div class="suggestions-item__name">
        {{ suggestion.name }}
      </div>

      <div class="suggestions-item__subname">
        {{ suggestion.nickName }}
      </div>

      <button class="suggestions-item__follow-cta">Follow</button>
    </div>
  </div>
</template>

<script>
export default {
  name: 'UserSuggestionItem',
  props: {
    suggestion: {
      type: Object,
      required: true,
    },
  },
};
</script>

<style lang="scss" scoped>
.suggestions-item {
  height: 200px;
  min-width: 180px;
  border: 1px solid $lighter-gray;
  padding: 10px;
  border-radius: 4px;
  margin-right: 20px;
  margin-bottom: 25px;

  &__header {
    display: flex;
    justify-content: flex-end;
  }

  &__remove {
    color: $dark-gray;
    font-weight: 500;
    cursor: pointer;
  }

  &__img-container {
    margin-bottom: 10px;
  }

  &__img {
    width: 54px;
    height: 54px;
    border-radius: 50%;
  }

  &__body {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    font-size: $font-size-2x;
  }

  &__name {
    font-weight: 600;
  }

  &__subname {
    color: $dark-gray;
  }

  &__follow-cta {
    border: 1px solid transparent;
    background-color: $main-color;
    margin-top: $font-size-3x;
    border-radius: 4px;
    color: $white;
    padding: 3px 40px;
    text-align: center;
  }
}
</style>
