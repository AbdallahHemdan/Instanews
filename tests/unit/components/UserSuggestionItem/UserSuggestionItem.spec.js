import { shallowMount } from '@vue/test-utils';
import UserSuggestionItem from './../../../../src/components/UserSuggestionItem/UserSuggestionItem.vue';

describe('UserSuggestionItem.vue', () => {
  let cmp;
  const suggestion = {
    img: require('./../../../../src/assets/Suggestions/1.jpg'),
    name: 'jomaoppa',
    nickName: 'Joma',
  };

  beforeEach(() => {
    cmp = shallowMount(UserSuggestionItem, {
      propsData: {
        suggestion,
      },
    });
  });

  describe('Testing props', () => {
    it('should render name of the user correctly', () => {
      expect(cmp.find('.suggestions-item__name').text()).toBe(suggestion.name);
    });

    it('should render nickname of the user correctly', () => {
      expect(cmp.find('.suggestions-item__subname').text()).toBe(suggestion.nickName);
    });

    it('should render image of the user correctly', () => {
      expect(cmp.find('.suggestions-item__img').attributes('src')).toBe(suggestion.img);
      expect(cmp.find('.suggestions-item__img').attributes('alt')).toBe(suggestion.name);
    });
  });

  describe('Testing inner text of elements', () => {
    it('should render close icon correctly', () => {
      expect(cmp.find('.suggestions-item__remove').text()).toBe('Ã—');
    });

    it('should render inner test of follow cta correctly', () => {
      expect(cmp.find('.suggestions-item__follow-cta').text()).toBe('Follow');
    });
  });

  describe('Snapshot testing', () => {
    it('should pass snapshot testing', () => {
      expect(cmp.element).toMatchSnapshot();
    });
  });
});
