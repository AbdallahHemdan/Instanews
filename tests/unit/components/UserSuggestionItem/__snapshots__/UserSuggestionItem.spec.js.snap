// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSuggestionItem.vue Snapshot testing should pass snapshot testing 1`] = `
<div
  class="suggestions-item"
>
  <div
    class="suggestions-item__header"
  >
    <div
      class="suggestions-item__remove"
    >
      Ã—
    </div>
  </div>
   
  <div
    class="suggestions-item__body"
  >
    <div
      class="suggestions-item__img-container"
    >
      <img
        alt="jomaoppa"
        class="suggestions-item__img"
        draggable="false"
        src=""
      />
    </div>
     
    <div
      class="suggestions-item__name"
    >
      
      jomaoppa
    
    </div>
     
    <div
      class="suggestions-item__subname"
    >
      
      Joma
    
    </div>
     
    <button
      class="suggestions-item__follow-cta"
    >
      Follow
    </button>
  </div>
</div>
`;
